File: lessons/222-nodejs-vs-ruby-performance/elixir-app/lib/app/device.ex
defmodule App.Device do
  use Ecto.Schema

  @derive {Jason.Encoder, only: [:id, :uuid, :mac, :firmware, :created_at, :updated_at]}
  schema "devices" do
    field(:uuid, :string)
    field(:mac, :string)
    field(:firmware, :string)
    field(:created_at, :utc_datetime_usec)
    field(:updated_at, :utc_datetime_usec)
  end
end

File: lessons/243-rust-vs-go-golang-performance-2025/rust-app/src/device.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Device<'a> {
    pub id: i32,
    pub uuid: &'a str,
    pub mac: &'a str,
    pub firmware: &'a str,
    pub created_at: &'a str,
    pub updated_at: &'a str,
}
